firmware-support/bittide-sys/src/uart/log.rs

```diff
@@ -44,15 +52,19 @@ impl log::Log for UartLogger {
             unsafe {
                 match &mut LOGGER.uart {
                     Some(l) => {
-                        writeln!(
-                            l,
-                            "{} | {}:{} - {}",
-                            record.level(),
-                            record.file().unwrap(),
-                            record.line().unwrap(),
-                            record.args()
-                        )
-                        .ok();
+                        if record.level() <= self.display_level {
+                            write!(l, "{} | ", record.level()).ok();
```

Comment:
I'm unsure what the purpose of the `.ok()`s here is. All it's doing is converting a `Result<T, E>` into an `Option<T>` by dropping the `E` value. If the use case here is to ignore the result, then I would suggest instead explicitly ignoring it by writing:
```
let _ = write!(l, "{} | ", record.level());
```